### DISCUSSION TITLE
Does jupyter have to have https?

### DISCUSSION THREAD

**Discussion by Pythoneryeah** [coder/code-server#6396] [Q&A]
1. Use https to access the code server of the remote host. jupyter is displayed normally.

2. When you access the code server of the remote host using http, jupyter cannot be displayed normally.

![image](https://github.com/coder/code-server/assets/37642828/a6ef48e0-ee9a-4f36-8d13-145e6caec539)

https doesn't work for me at all and takes a lot of extra time to configure. At the source level, is there any way to make http support jupyter functionality

**Comment by code-asher** [comment]
VS Code uses service workers, and browsers only allow service 
workers in a secure context, so there is nothing code-server can 
do about this restriction.

Now, maybe VS Code can be rewritten to avoid service workers in 
the first place but I have no idea how difficult that would be.

But, you may be able to disable the restriction in your browser's 
settings.
